cpu/esp32/vendor/esp-idf/heap/heap_trace.c

```diff
@@ -407,10 +407,13 @@ IRAM_ATTR void *__wrap_realloc(void *p, size_t size)
 
 IRAM_ATTR void *__wrap_calloc(size_t nmemb, size_t size)
 {
-    size = size * nmemb;
-    void *result = trace_malloc(size, 0, TRACE_MALLOC_DEFAULT);
+    size_t total_size;
```

Comment:
@maribu Is this change really necessary in this place? Or could we also put the check in a wrapper function in the RIOT port in `cpu/esp_common/syscalls.c`? The problem is that with each additional change to the vendor code, a later upgrade to a newer version becomes more difficult.

Reply:
I guess that would also be possible. It also could be that a future vendor code uses a newlib version that has the overflow check this does manually, so we could get rid of this completely. If you are in the process of updating the vendor code anyway, I would check if `tests/malloc` passes without this. If so, we don't need it anymore :)

Reply:
> If you are in the process of updating the vendor code anyway

Unfortunatly not. The vendor code used in RIOT port is from ESP-IDF version 3.1 which is more than 3 years old (by far too old) and still uses gcc 5.2.0.

The biggest problem is that it only supports the original ESP32 vairant. In the meantime, version we have ISP-IDF version 4.4 and a number of [different ESP32 variants that require newer versions of the ESP-IDF](https://github.com/espressif/esp-idf#esp-idf-release-and-soc-compatibility). A lot of ESP32 modules on the market use these ESP32 variants because they are mostly single core variants that are much cheaper. That is, we will have to upgrade the RIOT port to a newer ESP-IDF version, otherwise the ESP32 port will become more and more useless.

Unfortunatly, the upgrade to a newer version of the ESP-IDF is not trivial and requires a lot of work.